app/controllers/gift_requests_controller.rb

```diff
@@ -48,6 +48,19 @@ def catalog
     @gift_requests = @event.gift_requests.unreserved.order(created_at: :asc).page(params[:page] || 1)
   end
 
+
+  def reserve
+    @gift_request = GiftRequest.find(params[:id])
+    reservation = Reservation.new(gift_request_id: params[:id], donor_id: current_user.id)
+    if reservation.save
+      redirect_to wishlist_path(@gift_request.list.id)
+    else
+      # render "gift_request"
+      render status: 500
```

Comment:
Yeah, the best practice is to render the new/edit page with the semantic errors if the save fails.